wordpress-mobile/WordPress-Android
# IA Project: add post page creation sources tracking

Fixes #10744 

To better detail the post-list tracking category we created the following sub categories:

```
POST_FROM_SHORTCUT: post created from android shortcut long pressing on app icon
POST_FROM_POSTS_LIST: post created from the posts list fab
PAGE_FROM_PAGES_LIST: page created from the pages list fab
POST_FROM_NAV_BAR: post created from the navigation bar (until we will remove it)
POST_FROM_MY_SITE: post created from bottom sheet in my site screen
PAGE_FROM_MY_SITE: page created from bottom sheet in my site screen
POST_FROM_STATS: post created from stats empty view when no stats/post available yet
POST_FROM_NOTIFS_EMPTY_VIEW: post created from notifications unread page when empty
NO_DETAIL: all other cases container
```
Please note that currently `POST_FROM_MY_SITE` and `PAGE_FROM_MY_SITE` events are only available in zalpha flavor where the INFORMATION_ARCHITECTURE_AVAILABLE feature flag is activated, whereas the `POST_FROM_NAV_BAR` is available only in the vanilla (or wasabi); so to get all the above events you will need to use both the vanilla and the zalpha flavors.

## To test
1. Be sure that tracking is enabled in Me > App Settings > Privacy settings > Collect information ( 😁 )
2. Launch the app and trigger the above events
3. Check the logcat to see that you get the `editor_post_created` event with the `created_post_source_detail` property value like in the following example:

```
Tracked: editor_post_created, Properties: {"blog_id":157783356,"created_post_source_detail":"other","post_type":"post-from-my-site","created_post_source":"post-list","is_jetpack":false}
```

NOTE: if merged into develop this PR adds the tracking capability also without the feature flag activated. cc @osullivanchris 

PR submission checklist:
